### DISCUSSION TITLE
Network v15 OhSnap Upgrade and Lotus Releases Timeline

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#7898] [Announcement]
**More details on the OhSnap network upgrade can be found [here](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1922550)!**

Subscribe to notifications on https://status.filecoin.io/ for any additional network updates!
If you have any concerns or questions, leave a comment [here](https://github.com/filecoin-project/lotus/discussions/7935).

## Lotus releases 
**Note: This projected timeline may be subject to change based on testing and development.**

You will need to be on lotus v1.14.0 or ^ for the OhSnap upgrade. [lotus `v1.14.0`:](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0)**  is based on the last stable release, lotus `v1.13.2`, and includes **consensus-breaking changes and a couple of necessary improvements** for the Network v15 OhSnap upgrade. We recommend all lotus users to upgrade your node to this version as soon as the release is tagged. This upgrade requires new proof parameters, and we recommend all users to fetch the params ahead of the upgrade.

As mentioned in the[ OhSnap Notice](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1927638):
- lotus release candidates will include the temporary proof params for the community to test Snap Deal in the SnapNet.
  - Note: the temporary proof params are not available in master to prevent users from using the wrong params in mainnet.
- Final releases will be out **after** the trusted setup ceremony is completed and the final proof is available. New release candidates will be published before the calibration upgrade once the final proof is ready. 

## Tentative Releases and Network Upgrade Timelines

> Time below is in UTC.
- [x] 2022-01-12, lotus [v1.14.0-rc1](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0-rc1) (Note: This release candidate uses temporary proof params for Snap Deals and only sets upgrade epcoh for Butterfly-SnapNet, does not set the upgrade epochs for calibration and mainnet.)
- [x]  2022-01-17T19:00:00Z Butterfly-SnapNet upgrade @epoch 30262!
- [x] 2022-01-26, trusted setup ceremony ends
- [x] 2022-02-08, release candidates with final proof parameters 
- [x] 2022-02-10T19:23:00Z, calibration upgrade  @epoch 682006.
- [x] 2022-02-17, [lotus v1.14.0](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0) 
- [x] 2022-03-01T15:00:00Z,  mainnet upgrade  @1594680

---
# Note: ⚠️ CONTEXT BELOW WAS THE ORIGINAL POST, archived on Feb 7th, 2022**

**More details on the OhSnap network upgrade can be found [here](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1922550)!**

Subscribe to notifications on https://status.filecoin.io/ for any additional network updates!
If you have any concerns or questions, leave a comment [here](https://github.com/filecoin-project/lotus/discussions/7935).

## Lotus releases 
**Note: This projected timeline may be subject to change based on testing and development.**

You will need to be on lotus v1.14.0 or ^ for the OhSnap upgrade. We will be publishing two lotus releases before this upgrade:
- **[lotus `v1.14.0`:](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0)** It is based on the last stable release, lotus `v1.13.2`, and includes **consensus-breaking changes and a couple of necessary improvements** for the Network v15 OhSnap upgrade. We recommend ecosystem teams like node operators, exchanges, and so on to upgrade your node to this version as soon as the release is tagged.
- [**lotus `v1.15.0`:**](https://github.com/filecoin-project/lotus/releases/tag/v1.15.0)  This release will include everything in `v1.14.0` plus the latest functionality and improvements, including data transfer perf improvements, sealing pipeline improvements, and so on. **We highly recommend users like storage providers, data brokers, and so on to upgrade your nodes and all systems to this release once it's out!**

As mentioned in the[ OhSnap Notice](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1927638):
- lotus release candidates will include the temporary proof params for the community to test Snap Deal in the SnapNet.
  - Note: the temporary proof params are not available in master to prevent users from using the wrong params in mainnet.
- Final releases will be out **after** the trusted setup ceremony is completed and the final proof is available. New release candidates will be published before the calibration upgrade once the final proof is ready. 

## Tentative Releases and Network Upgrade Timelines

> Time below is in UTC.
- [x] 2022-01-12, lotus [v1.14.0-rc1](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0-rc1) (Note: This release candidate uses temporary proof params for Snap Deals and only sets upgrade epcoh for Butterfly-SnapNet, does not set the upgrade epochs for calibration and mainnet.)
- [x]  2022-01-17T19:00:00Z Butterfly-SnapNet upgrade @epoch 30262!
- [x] 2022-01-26, trusted setup ceremony ends
- [x] 2022-02-01, release canadiates with final proof paramters
- [x]  2022-02-08, calibration upgrade
- [x] 2022-02-15, [lotus v1.14.0] 
- [x] 2022-03-01,  mainnet upgrade 

> Note: ⚠️ END

**Comment by jennijuju** [comment]
## Updates About OhSnap and stuff

First of all, I wanna thank @rjan90 , @benjaminh83 and others who participated in SnapNet testing over the past two weeks. With the bugs you caught and feedbacks you provided, the rest of the lotus community may have a better snap user experience when it lands in Mainnet! 💙 

A couple things:
- We are almost ready for the final v1.14.0, two main remaining items are captured [here](https://github.com/filecoin-project/lotus/pull/8020) and will be delivered later this week.
- Trusted setup was completed successfully and proofs are going through final verifications. SnapNet was reset and upgraded last Friday with the final params.
- We will cut v1.14.0-rc6 soon, with the final params and the calibration OhSnap upgrade params. Calibration upgrade is scheduled on **Feb 10th, 19:23 UTC, epoch 682006**.
- It was lotus team's top priority to provide SPs a as smooth user experience to upgrade existing CC sectors via SNAP as possible, and we believed it should be considered as a deliverable goal for v1.14.0 (the mandatory release for ntwk v15). Therefore, majority of the SnapDeal work has been included in v1.14.0. 
  - That being said, the lotus team had a reduced dev capacity in Dec & Jan and we had some delays in delivering the first rc for v1.15.0(which was mentioned in the original announcement). It includes improvements wrt libp2p, graph sync and so on. At this point, following the lotus release flow, it's likely we won't be able to ship the final v1.15.0 release before the mainnet OhSnap upgrade(Mar 1st). Thus, we recommend v1.14.0 for all users for this upgrade.
  - v1.15.0-rc1 is aimed to be cut on Feb 8th, following the regular lotus release testing flow!
- There will be an AMA on Feb 17th, and the lotus team will be there answering questions about deal making and snapdeals!

**Comment by jennijuju** [comment]
## OhSnap READY

[lotus v1.14.0](https://github.com/filecoin-project/lotus/releases/tag/v1.14.0) is out (commit https://github.com/filecoin-project/lotus/commit/3bb6efe0db9ea9fed8770387636028fef078f3b5 ) , please read the release note for more details and reminders! The main net upgrade will be on **2022-03-01T15:00:00Z, @ epoch 1594680**!

Small nit being.. we had a typo in the **comment** of v1.14.0 release about the main net upgrade **date**[ here](https://github.com/filecoin-project/lotus/blob/c8edd5bd52a43186564792b96d344fdb250ef6d3/build/params_mainnet.go#L70), the current date is **March 1st**. We will fix this comment in v1.14.1 and ^ later on.

If you have any question about the release, leave a comment here https://github.com/filecoin-project/lotus/discussions/8145.
If you have any question about the upgrade in general, join the discussion here: https://github.com/filecoin-project/lotus/discussions/7935

**Comment by jennijuju** [comment]
## OHSNAP was upgraded SUCCESSFULLY!

**Comment by arajasek** [comment]
For anyone who wants to check if they're on the right chain, run:

`lotus state --tipset @1594685 compute-state`

If the result is `computed state cid:  bafy2bzacebmcrunsuog3j44ljwjeai3rtb6ugf6jgkz4ircchicevx6pnk2sy`, you're good!
